rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,633 @@
+          required: false
+          schema:
+            type: string
+          example: ?parent=accounting.tax
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ListNamespacesResponse'
+        '401':
+          description: Unauthorized
+        '404':
+          description: Not Found (Parent namespace does not exist)
+    post:
+      tags:
+        - Catalog API
+      summary: Create a namespace
+      description: Create a namespace, with an optional set of properties. The server might also add properties, such as last_modified_time etc.
+      operationId: createNamespace
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateNamespaceRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+        '401':
+          description: Unauthorized
+        '409':
+          $ref: '#/components/responses/NamespaceAlreadyExistsResponse'
+
+  /namespaces/{namespace}:
+    parameters:
+      - name: namespace
+        in: path
+        required: true
+        schema:
+          type: string
+    get:
+      tags:
+        - Catalog API
+      summary: Load the metadata properties for a namespace
+      operationId: loadNamespaceMetadata
+      description: Return all stored metadata properties for a given namespace
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GetNamespaceResponse'
+        '404':
+          $ref: '#/components/responses/NoSuchNamespaceResponse'
+    head:
+      tags:
+        - Catalog API
+      summary: Check if a namespace exists
+      operationId: namespaceExists
+      description: Check if a namespace exists.
+      responses:
+        '200':
+          description: OK - Namesapce exists
+        '401':
+          description: Unauthorized
+        '404':
+          description: Not Found
+    delete:
+      tags:
+        - Catalog API
+      summary: Drop a namespace from the catalog. Namespace must be empty.
+      operationId: dropNamespace
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/IcebergResponseObject'
+              example: { "data": { "dropped": true } }
+        '401':
+          description: Unauthorized
+        '404':
+          description: Not Found
+
+
+  /namespaces/{namespace}/properties:
+    parameters:
+      - name: namespace
+        in: path
+        required: true
+        schema:
+          type: string
+    post:
+      tags:
+        - Catalog API
+      summary: Set or remove properties on a namespace
+      operationId: setProperties / removeProperties
```

Comment:
I had not heard that before. Do you have any source indicating that?

I updated this to use the slash because this one route is handling both `setPrpoperties` call and `removeProperties` call: https://github.com/apache/iceberg/blob/59a76da2ad78b3fabed024baa359e931d239ddc9/api/src/main/java/org/apache/iceberg/catalog/SupportsNamespaces.java#L107-L129

editor.swagger.io did not complain (and it seems to complain about a number of other things), but if what you're is true, then probably the operationId could be `setOrRemoveProperties` or something.

Reply:
I guess it's more a suggestion than a specification? But I took it from https://swagger.io/specification/#operation-object

> Unique string used to identify the operation. The id MUST be unique among all operations described in the API. The operationId value is case-sensitive. Tools and libraries MAY use the operationId to uniquely identify an operation, therefore, it is RECOMMENDED to follow common programming naming conventions.

Reply:
Ah ok. I see. I used it in editor. swagger.io and it didn't complain about it. But I can change it to updateProperties, since it's now really a mixture of both.

But I do like that it currently maps to existing API names.